functions/Copy-SqlSsisCatalog.ps1

```diff
@@ -0,0 +1,420 @@
+                foreach ($e in $envs) {
+                    $remove.Environments[$e].Drop()
+                }
+                $projs = $remove.Projects.Name
+                foreach ($p in $projs) {
+                    $remove.Projects[$p].Drop()
+                }
+                $remove.Drop()
+                $destinationCatalog.Alter()
+                $destinationCatalog.Refresh()
+            }
+            Write-Output "Creating folder $Folder."
+            $destFolder = New-Object "$ISNamespace.CatalogFolder" ($destinationCatalog, $Folder, $Description)
+            $destFolder.Create()
+            $destFolder.Alter()
+            $destFolder.Refresh()
+        }
+        
+        Function Create-Environment {
+            param(
+                [String]$Folder,
+                [String]$Environment,
+                [Switch]$Force
+            )
+            $envDestFolder = $destinationFolders | Where-Object { $_.Name -eq $Folder }
+            if ($force) {
+                $envDestFolder.Environments[$Environment].Drop()
+                $envDestFolder.Alter()
+                $envDestFolder.Refresh()
+            }
+            $srcEnv = ($sourceFolders | Where-Object { $_.Name -eq $Folder }).Environments[$Environment]
+            $targetEnv = New-Object "$ISNamespace.EnvironmentInfo" ($envDestFolder, $srcEnv.Name, $srcEnv.Description)
+            foreach ($var in $srcEnv.Variables) {
+                if ($var.Value.ToString() -eq "") { 
+                    $finalValue= ""
+                }
+                else {
+                    $finalValue = $var.Value
+                }
+                $targetEnv.Variables.Add($var.Name, $var.Type, $finalValue, $var.Sensitive, $var.Description)
+            }
+            Write-Output "Creating environment $Environment."
+            $targetEnv.Create()
+            $targetEnv.Alter()
+            $targetEnv.Refresh()
+        }
+
+        Function Create-SSISDBCatalog {
+            Write-Output "SSISDB Catalog requires a password."
+            $pass1 = Read-Host "Enter a password" -AsSecureString
+            $plainTextPass1 = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($pass1))
+            $pass2 = Read-Host "Re-enter password" -AsSecureString
+            $plainTextPass2 = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($pass2))
+            if ($plainTextPass1 -ne $plainTextPass2) {
+                Throw "Validation error, passwords entered do not match."
+            }
+            else {
+                $catalog = New-Object "$ISNamespace.Catalog" ($destinationSSIS, "SSISDB", $plainTextPass1)  
+                $catalog.Create()
+                $catalog.Refresh()
+            }
+        }
+    }
+    PROCESS
+    {
+        if (!$sourceCatalog) {
+            Write-Error "The source SSISDB catalog does not exist."
+            exit
```

Comment:
Sorry, a couple more that I missed from yesterday. Write-Error is non terminating errors and throw is for terminating errors. In this case, you'd want to  

throw "The source SSISDB catalog does not exist."

Also, please change all exits to returns or throws. Exit can close a PowerShell window and is a bit too harsh. Return is a better alternative.